stock_picking_auto_create_lot/models/stock_move_line.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+from odoo.fields import first
+
+
+class StockMoveLine(models.Model):
+
+    _inherit = "stock.move.line"
+
+    def _prepare_auto_lot_values(self):
+        """
+        Prepare multi valued lots per line to use multi creation.
+        """
+        self.ensure_one()
+        return {"product_id": self.product_id.id, "company_id": self.company_id.id}
+
+    def set_lot_auto(self):
+        """
+            Create lots using create_multi to avoid too much queries
+            As move lines were created by product or by tracked 'serial'
+            products, we apply the lot with both different approaches.
+        """
+        values = []
+        production_lot_obj = self.env["stock.production.lot"]
+        for line in self:
+            values.append(line._prepare_auto_lot_values())
+        lots = production_lot_obj.create(values)
+        for line in self:
+            lot = first(lots.filtered(lambda l: l.product_id == line.product_id))
```

Comment:
@rousseldenis IMO you should avoid to filter on each loop by creating a dict of lot by product_id before the loop

Reply:
Indeed, I'm currently trying to improve this. Not finished

Reply:
@lmignon Done